ui/src/app/workflows/components/submit-workflow-panel.tsx

```diff
@@ -31,6 +34,7 @@ export function SubmitWorkflowPanel(props: Props) {
     const {navigation} = useContext(Context);
     const [entrypoint, setEntrypoint] = useState(workflowEntrypoint);
     const [parameters, setParameters] = useState<Parameter[]>([]);
+    const [namespace, setNamespace] = useState(props.namespace);
```

Comment:
So this is not entirely correct to have `props.namespace` be the initial state. For a `WorkflowTemplate`, it will _always_ be `props.namespace`, and for a `ClusterWorkflowTemplate`, it will always be the `namespace` state.

This would be particularly relevant in cases where this panel is off-screen but still rendered; the next time it's on-screen, the prop would not be used.
That's normally not how this component is used, but that is why the correctness matters if it were to be used that way.

Also this should be ordered after the `useContext` instead of in between the two parameter state variables for better organization

Reply:
You're right. I pushed 59dbea3690b1334d8f9b4cfc1e04ef60cc679141 to lift the state up to `<ClusterWorkflowTemplateDetails>`, which is simpler than handling it here